engine/src/Interpreter/interpreter_ops.cuh

```diff
@@ -664,11 +664,17 @@ private:
 			}
 			bool left_valid = getColumnValid(row_valids, left_position);
 
-			if(oper == operator_type::BLZ_IS_NULL) {
+			if (oper == operator_type::BLZ_IS_NOT_TRUE) {
+				left_valid = (left_value == false) ? true : false;
```

Comment:
This logic does not look right. Please verify.
Also, if this logic is not correct, how is this passing e2e tests?

Reply:
same questions about BLZ_IS_NOT_FALSE

Reply:
The way this code is written, it look like its either true or null, and never false. I think its actually either true or false and never null.

Reply:
For a boolean column, `left_value` is also `boolean`. 
When the `value_i` is `null` or `false`, then `left_value` will be`False`.

Reply:
The next column diagram illustrates the right bool values for `left_value` and `left_valid`  when the column is boolean. 
BOOL_COL  | left_value  |   left_valid
null         |   False	   |	 False
True	     |	True	   |	 True
False		 |	False	   |	 True
null		 |	False	   |	 False
False		 |	False	   |	 True
True		 |	True	   |	 True

Reply:
For a `boolean` column, `left_value` is also boolean.
When the `value_i` is `null` or `false`, then `left_value` will be `False`.

The next column diagram illustrates the right bool values for left_value and left_valid when the column is boolean.
```
BOOL_COL | left_value | left_valid
null  | False | False
True  | True  | True
False | False | True
null  | False | False
False | False | True
True  | True  | True
```

Reply:
Ok, to clarify. This is the logic table as I undestand it, and  i think we agree that this is what it should be:

BOOL Col | left_value | left_valid | IS_NOT_TRUE | IS_NOT_FALSE | IS_TRUE
null            | N/A          | FALSE    | TRUE                 | TRUE                 | FALSE
TRUE       | TRUE       | TRUE     | FALSE                | TRUE                 | TRUE
FALSE      |     FALSE | TRUE     | TRUE                  | FALSE                | FALSE

This is your code:
```
if (oper == operator_type::BLZ_IS_NOT_TRUE) {
    left_valid = (left_value == false) ? true : false;
    store_data_in_buffer(static_cast<int64_t>(true), buffer, output_position);
```
Notice that you are always setting the value to true. And the valid is depending on the left_value. 
Instead it should be:
```
if (oper == operator_type::BLZ_IS_NOT_TRUE) {
     bool val = (left_valid == true && left_value == true) ? false : true;
    left_valid = true;
    store_data_in_buffer(static_cast<int64_t>(val), buffer, output_position);
```
`BLZ_IS_NOT_FALSE` also needs to be fixed